### ISSUE TITLE
fish_config ignores system default browser

### ISSUE THREAD

**Issue by cabello** [fish-shell/fish-shell#4210]
While using macOS Sierra 10.12.5 and iTerm2 Build 3.0.15, when I run `fish_config`, the configuration does not open in my default browser (Chrome) and it opens in Firefox instead. The same behaviour happens in Terminal app.

```
➤ fish --version
fish, version 2.5.0
➤ uname -a
Darwin ... 16.6.0 Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64 x86_64
➤ echo $TERM
xterm-256color
```

**Comment by faho** [comment]
Duplicate of #4035. Note that the underlying issue is in macOS. We've added a workaround that will be included in the next fish release.

**Comment by zanchey** [comment]
We didn't take that for 2.6.0, and I think we'll back it out for 2.7.0; the correct thing to do here is upgrade OS X (which contains security fixes anyway).

**Comment by faho** [comment]
Has there been a macOS release that fixes it? Because otherwise we should keep it, since AFAIK it doesn't break anything, only make it a bit less nice.

**Comment by zanchey** [comment]
No, but it will be released by the time 2.7.0 is out (I'm holding off until it's generally available in case we get there first).
